apps/src/lib/levelbuilder/code-docs-editor/ProgrammingExpressionEditor.jsx

```diff
@@ -13,15 +13,29 @@ import $ from 'jquery';
 import color from '@cdo/apps/util/color';
 
 function useProgrammingExpression(initialProgrammingExpression) {
-  const [programmingExpression, setProgrammingExpression] = useState(
-    initialProgrammingExpression
+  const initializeProgrammingExpression = programmingExpression => {
+    const copiedExpression = {...programmingExpression};
+    // We remove id and key from state as they should not be modified
+    delete copiedExpression.id;
+    delete copiedExpression.key;
+    if (copiedExpression.examples) {
```

Comment:
I think it might be worth a comment here (and in the other classes) explaining why we need to set the key.

One note -- combined with the change to call this on reset, this means that after a call to reset all of the keys will change.  Just wanted to make sure that was ok?

Reply:
I think it will already do that because in reality this function is called when we get new data from the server (which wouldn't have these keys).